### PR TITLE
Index anvil2 in anvilprod (#5489)

### LINKED ISSUE #5489: Index anvil2 in anvilprod
The wranglers have released 50 more datasets to be indexed by Azul. 

Please see the `anvil2` column in the `AnVIL Prod` tab of the "[TDR Dataset and snapshots](https://docs.google.com/spreadsheets/d/1yYiB3TrX0wvLCHvtWY1GTZZcAf5MmMREgn52OKCHsqs/edit?usp=sharing)" spreadsheet.

These are datasets from a variety of studies.

[Slack thread](https://ucsc-gi.slack.com/archives/C03TPJS54DC/p1693341779811629)

### PR DESCRIPTION
**Pull request by dsotirho-ucsc**
Connected issues: #5489

### Author

- [x] PR is a draft
- [x] Target branch is `develop`
- [x] Name of PR branch matches `issues/<GitHub handle of author>/<issue#>-<slug>`
- [x] PR title references all connected issues
- [x] PR title matches<sup>1</sup> that of a connected issue <sub>or comment in PR explains why they're different</sub>
- [x] For each connected issue, there is at least one commit whose title references that issue
- [x] PR is connected to all connected issues via ZenHub
- [x] PR description links to connected issues
- [x] Added `partial` label to PR <sub>or this PR completely resolves all connected issues</sub>

<sup>1</sup> when the issue title describes a problem, the corresponding PR
title is `Fix: ` followed by the issue title

### Author (reindex, API changes)

- [x] Added `r` tag to commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR <sub>or this PR does not require reindexing</sub>
- [x] PR and connected issue are labeled `API` <sub>or this PR does not modify a REST API</sub>
- [x] Added `a` (`A`) tag to commit title for backwards (in)compatible changes <sub>or this PR does not modify a REST API</sub>
- [x] Updated REST API version number in `app.py` <sub>or this PR does not modify a REST API</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `base` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Added `chained` label to this PR <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR <sub>or this PR does not require upgrading</sub>

### Author (operator tasks)

- [x] Added checklist items for additional operator tasks <sub>or this PR does not require additional tasks</sub>

### Author (hotfixes)

- [x] Added `F` tag to main commit title <sub>or this PR does not include permanent fix for a temporary hotfix</sub>
- [x] Reverted the temporary hotfixes for any connected issues <sub>or the `prod` branch has no temporary hotfixes for any connected issues</sub>

### Author (before every review)

- [x] Rebased PR branch on `develop`, squashed old fixups
- [x] Ran `make requirements_update` <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR <sub>or this PR does not touch requirements*.txt</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

### Peer reviewer (after requesting changes)

Uncheck the *Author (before every review)* checklists.

### Peer reviewer (after approval)

- [x] PR is not a draft
- [x] Ticket is in *Review requested* column
- [x] Requested review from primary reviewer
- [x] Assigned PR to primary reviewer

### Primary reviewer (after requesting changes)

Uncheck the *before every review* checklists. Update the `N reviews` label.

### Primary reviewer (after approval)

- [x] Actually approved the PR
- [x] Labeled connected issues as `demo` or `no demo`
- [x] Commented on connected issues about demo expectations <sub>or all connected issues are labeled `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] `N reviews` label is accurate
- [x] Moved ticket to *Approved* column
- [x] Assigned PR to current operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear <sub>or all connected issues are labeled `no demo`</sub>
- [x] PR has checklist items for upgrading instructions <sub>or PR is not labeled `upgrade`</sub>
- [x] Squashed PR branch and rebased onto `develop`
- [x] Sanity-checked history
- [x] Pushed PR branch to GitHub
- [x] Pushed PR branch to GitLab `dev` and added `sandbox` label <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed PR branch to GitLab `anvilprod` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `sandbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `anvilbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Reviewed build logs for anomalies in `hammerbox` deployment <sub>or PR is labeled `no sandbox`</sub>
- [x] Deleted unreferenced indices in `sandbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvilbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `hammerbox` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Started reindex in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Started reindex in `hammerbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `anvilbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `hammerbox` <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Title of merge commit starts with title from this PR
- [x] Added PR reference to merge commit title
- [x] Added commit title tags to merge commit title
- [x] Moved connected issues to Merged column in ZenHub
- [x] Pushed merge commit to GitHub

### Operator (chain shortening)

- [x] Changed the target branch of the blocked PR to `develop` <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `chained` label from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the blocking relationship from the blocked PR <sub>or this PR is not labeled `base`</sub>
- [x] Removed the `base` label from this PR <sub>or this PR is not labeled `base`</sub>

### Operator (after pushing the merge commit)

- [x] Pushed merge commit to GitLab `dev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed merge commit to GitLab `anvildev` <sub>or PR is labeled `no sandbox`</sub>
- [x] Pushed merge commit to GitLab `anvilprod` <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passes on GitLab `dev`<sup>1</sup>
- [x] Reviewed build logs for anomalies on GitLab `dev`<sup>1</sup>
- [x] Build passes on GitLab `anvildev`<sup>1</sup>
- [x] Reviewed build logs for anomalies on GitLab `anvildev`<sup>1</sup>
- [x] Build passes on GitLab `anvilprod`<sup>1</sup>
- [x] Reviewed build logs for anomalies on GitLab `anvilprod`<sup>1</sup>
- [x] Deleted PR branch from GitHub
- [x] Deleted PR branch from GitLab `dev`
- [x] Deleted PR branch from GitLab `anvildev`
- [x] Deleted PR branch from GitLab `anvilprod`

<sup>1</sup> When pushing the merge commit is skipped due to the PR being
labelled `no sandbox`, the next build triggered by a PR whose merge commit *is*
pushed determines this checklist item.

### Operator (reindex)

- [x] Deleted unreferenced indices in `dev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvildev` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Deleted unreferenced indices in `anvilprod` <sub>or this PR does not remove catalogs or otherwise causes unreferenced indices </sub>
- [x] Started reindex in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Started reindex in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Started reindex in `anvilprod` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `dev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `anvildev` <sub>or this PR does not require reindexing</sub>
- [x] Checked for and triaged indexing failures in `anvilprod` <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `dev` deployment <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `anvildev` deployment <sub>or this PR does not require reindexing</sub>
- [x] Emptied fail queues in `anvilprod` deployment <sub>or this PR does not require reindexing</sub>

### Operator

- [x] Unassigned PR

### Author

- [x] Contact the Broad about problematic snapshots

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Review by achave11-ucsc — DISMISSED** [review]

**Comment by achave11-ucsc** [comment]
@dsotirho-ucsc please consider determining if it's necessary for the first `anvil` catalog to remain in ES, it's likely that the DB no longer uses it.

File: deployments/anvilprod/environment.py [hunk 1]
```diff
@@ -131,6 +131,73 @@ def mkdict(previous_catalog: dict[str, str],
     mksrc('datarepo-1cb73890', 'ANVIL_cmg_broad_heart_ware_wes_20221215_ANV5_202304242145', 40),
 ]))
 
+anvil2_sources = mkdict({}, 50, mkdelta([
```
Comment (nadove-ucsc) [inline deployments/anvilprod/environment.py]: Question: is omitting the `anvil1` source intentional? For HCA, we usually include the sources from the previous catalog(s) when we add a new one.

**Review by nadove-ucsc — CHANGES_REQUESTED** [review]
All changes to the `anvilprod` deployment should be also be applied to the `hammerbox` deployment.

File: deployments/anvilprod/environment.py (hunk 1 above)
Comment (dsotirho-ucsc) [reply deployments/anvilprod/environment.py]: No, the omission was not intentional. Good catch.

**Review by nadove-ucsc — DISMISSED** [review]

**Comment by nadove-ucsc** [comment]
PR does not need to be built on `sandbox` or `anvilbox`, but _does_ need to be built on `hammerbox`.

**Comment by dsotirho-ucsc** [comment]
@hannes-ucsc: "Included Noa's suggested change to add the prefix and also remove the `anvil` catalog as a first commit."

**Review by nadove-ucsc — DISMISSED** [review]

**Comment by dsotirho-ucsc** [comment]
IT on hammerbox failed with messages left in the fail queues, GitLab Job [11171](https://gitlab.prod.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/11171)

This [CloudWatch query](https://us-east-1.console.aws.amazon.com/cloudwatch/home?region=us-east-1#logsV2:logs-insights$3FqueryDetail$3D~(end~'2023-09-06T20*3a55*3a16.960Z~start~'2023-09-06T15*3a42*3a59.610Z~timeType~'ABSOLUTE~tz~'Local~editorString~'fields*20*40timestamp*2c*20*40requestId*2c*20notification.bundle_fqid.source.spec*2c*20*40message*0a*7c*20filter*20*40message*20like*20*2f*28*3fi*29too*20large*2f*0a*7c*20sort*20*40requestId*2c*20*40timestamp*20asc*0a*7c*20limit*20200~queryId~'4630c07890eeb91c-e5c693b2-4ce8258-2dd1f59d-7962f2c6f12c19bc20459b10~source~(~'*2faws*2flambda*2fazul-indexer-hammerbox~'*2faws*2flambda*2fazul-indexer-hammerbox-aggregate~'*2faws*2flambda*2fazul-indexer-hammerbox-contribute~'*2faws*2flambda*2fazul-indexer-hammerbox-contribute_retry))) shows a problem caused by two of the new sources:
- tdr:datarepo-cddcc424:snapshot/ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050:13/0
- tdr:datarepo-75723148:snapshot/ANVIL_NIMH_Broad_WGSPD1_McCarroll_Pato_GRU_10XLRGenomes_20230331_ANV5_202304242054:0/0

These errors are repeated a few times in the logs:
```
GoogleAPICallError: 413 POST https://bigquery.googleapis.com/bigquery/v2/projects/platform-anvil-prod/jobs?prettyPrint=false: <!DOCTYPE html>
<html lang=en>
  <meta charset=utf-8>
  <meta name=viewport content="initial-scale=1, minimum-scale=1, width=device-width">
  <title>Error 413 (Request Entity Too Large)!!1</title>
  <style>
    *{margin:0;padding:0}html,code{font:15px/22px arial,sans-serif}html{background:#fff;color:#222;padding:15px}body{margin:7% auto 0;max-width:390px;min-height:180px;padding:30px 0 15px}* > body{background:url(//www.google.com/images/errors/robot.png) 100% 5px no-repeat;padding-right:205px}p{margin:11px 0 22px;overflow:hidden}ins{color:#777;text-decoration:none}a img{border:0}@media screen and (max-width:772px){body{background:none;margin-top:0;max-width:none;padding-right:0}}#logo{background:url(//www.google.com/images/branding/googlelogo/1x/googlelogo_color_150x54dp.png) no-repeat;margin-left:-5px}@media only screen and (min-resolution:192dpi){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat 0% 0%/100% 100%;-moz-border-image:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) 0}}@media only screen and (-webkit-min-device-pixel-ratio:2){#logo{background:url(//www.google.com/images/branding/googlelogo/2x/googlelogo_color_150x54dp.png) no-repeat;-webkit-background-size:100% 100%}}#logo{display:inline-block;height:54px;width:150px}
  </style>
  <a href=//www.google.com/><span id=logo aria-label=Google></span></a>
  <p><b>413.</b> <ins>That’s an error.</ins>
  <p>Your client issued a request that was too large.
 <script>
  (function() { var c=function(a,d,b){a=a+"=deleted; path="+d;null!=b&&(a+="; domain="+b);document.cookie=a+"; expires=Thu, 01 Jan 1970 00:00:00 GMT"};var g=function(a){var d=e,b=location.hostname;c(d,a,null);c(d,a,b);for(var f=0;;){f=b.indexOf(".",f+1);if(0>f)break;c(d,a,b.substring(f+1))}};var h;if(4E3<unescape(encodeURI(document.cookie)).length){for(var k=document.cookie.split(";"),l=[],m=0;m<k.length;m++){var n=k[m].match(/^\s*([^=]+)/);n&&l.push(n[1])}for(var p=0;p<l.length;p++){var e=l[p];g("/");for(var q=location.pathname,r=0;;){r=q.indexOf("/",r+1);if(0>r)break;var t=q.substring(0,r);g(t);g(t+"/")}"/"!=q.charAt(q.length-1)&&(g(q),g(q+"/"))}h=!0}else h=!1;
h&&setTimeout(function(){if(history.replaceState){var a=location.href;history.replaceState(null,"","/");location.replace(a)}},1E3); })();
</script>
 <ins>That’s all we know.</ins>
```

```
Worker failed to handle message {'action': 'add', 'notification': {'transaction_id': 'bf8c3c5f-8a02-4899-b317-4016385b53e1', 'bundle_fqid': {'uuid': '13ea653e-395e-a804-bac6-857b7a70cafc', 'version': '2022-06-01T00:00:00.000000Z', 'source': {'id': '69007912-8f45-4925-a857-57a5c09ca536', 'spec': 'tdr:datarepo-cddcc424:snapshot/ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050:13/0'}, 'entity_type': 'biosample'}}, 'catalog': 'anvil2-it'}.
Traceback (most recent call last):
  File "/var/task/azul/indexer/index_controller.py", line 179, in contribute
    contributions = self.transform(catalog, notification, delete)
  File "/var/task/azul/indexer/index_controller.py", line 213, in transform
    bundle = service.fetch_bundle(catalog, bundle_fqid)
  File "/var/task/azul/indexer/index_service.py", line 180, in fetch_bundle
    return plugin.fetch_bundle(bundle_fqid)
  File "/var/task/azul/plugins/repository/tdr.py", line 208, in fetch_bundle
    bundle = self._emulate_bundle(bundle_fqid)
  File "/var/task/azul/plugins/repository/tdr_anvil/__init__.py", line 249, in _emulate_bundle
    return self._primary_bundle(bundle_fqid)
  File "/var/task/azul/plugins/repository/tdr_anvil/__init__.py", line 287, in _primary_bundle
    rows = self._retrieve_entities(source.spec, entity_type, typed_keys)
  File "/var/task/azul/plugins/repository/tdr_anvil/__init__.py", line 644, in _retrieve_entities
    rows = self._run_sql(f'''
  File "/var/task/azul/plugins/repository/tdr.py", line 221, in _run_sql
    return self.tdr.run_sql(query)
  File "/var/task/azul/terra.py", line 522, in run_sql
    job: QueryJob = bigquery.query(query)
  File "/opt/python/google/cloud/bigquery/client.py", line 3390, in query
    future = do_query()
  File "/opt/python/google/cloud/bigquery/client.py", line 3367, in do_query
    query_job._begin(retry=retry, timeout=timeout)
  File "/opt/python/google/cloud/bigquery/job/query.py", line 1298, in _begin
    super(QueryJob, self)._begin(client=client, retry=retry, timeout=timeout)
  File "/opt/python/google/cloud/bigquery/job/base.py", line 510, in _begin
    api_response = client._call_api(
  File "/opt/python/google/cloud/bigquery/client.py", line 782, in _call_api
    return call()
  File "/opt/python/google/api_core/retry.py", line 283, in retry_wrapped_func
    return retry_target(
  File "/opt/python/google/api_core/retry.py", line 190, in retry_target
    return target()
  File "/opt/python/google/cloud/_http/__init__.py", line 494, in api_request
    raise exceptions.from_http_response(response)
google.api_core.exceptions.BadRequest: 400 POST https://bigquery.googleapis.com/bigquery/v2/projects/platform-anvil-prod/jobs?prettyPrint=false: The query is too large. The maximum standard SQL query length is 1024.00K characters, including comments and white space characters.
Location: None
```

**Comment by hannes-ucsc** [comment]
The obvious solution would be to batch the queries but first I'd like us to investigate the graph structure a bit more. What are the inputs and outputs to/from those activities? Do the activities contain meaningful data?

**Comment by nadove-ucsc** [comment]
The number of activities doesn't make any sense. Bundle `13ea653e-395e-a804-bac6-857b7a70cafc` from `datarepo-cddcc424.ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050` contains a total of 106,786 entities, 104,909 (98.2%) of which are activities. In our bundle model, every activity is linked to a positive number of inputs and a positive number of outputs, none of which may be activities. Therefore, the only way it could be mathematically possible for the number of activity entities to exceed the number of non-activity entities in the bundle is for there to be multiple redundant activities that connect the same sets of inputs and outputs.

**Comment by nadove-ucsc** [comment]
There is indeed a lot of structural redundancy in the activities. There are only 935 distinct values for `anvil_activity.generated_file_id` in the entire snapshot. Of these files, 374 have only one activity that generates them. The other 561 are where the redundancy occurs. 

These 561 files can be grouped into three sets of 187 files each. Each file is generated by 187 distinct activities, each with a single file input. All files in the same group share the same set of 187 inputs via a total of 34,969 activities (1 input and 1 output per activity). Across all three groups, this accounts for the vast majority of activities in the snapshot. All of these activities are from the generic `anvil_activity` table, so there is no metadata associated with the activities themselves besides the `activity_type` field, which is the same ("Indexing") for all activities discussed in this paragraph.

This seems like a flaw in the structure of the snapshot to me. For a start, it doesn't make real-world sense to me for a file to be generated by more than 1 activity. Objectively, because the `used_file_id` field is an array, each set of 187 single-input activities that generate the same file could be consolidated into a single activity with 187 inputs. From Azul's perspective, we could reduce things even further by having a single activity per group with 187 inputs and 187 outputs, but this may not reflect the underlying procedures.

**Comment by dsotirho-ucsc** [comment]
@hannes-ucsc: "I agree with the assessment that this is likely a bug in the snapshot. We will remove all affected snapshots from the `anvil2` catalog and start a conversation with Nate about this issue on Slack. Before we do that we need to determine which snapshots are affected by looking at the ratio between number of activities and number of files in each new snapshot. @nadove-ucsc to do so."

**Comment by nadove-ucsc** [comment]
```
$ head -15 afratio.txt 
Snapshot name                                                                                    #files       #activities  #a/#f
-----
ANVIL_eMERGE_HMB_IRB_PUB_eMERGEseq_20230130_ANV5_202304271622                                    0            478          inf
ANVIL_eMERGE_GRU_NPU_eMERGEseq_20230130_ANV5_202304271617                                        0            5907         inf
ANVIL_eMERGE_GRU_IRB_PUB_NPU_eMERGEseq_20230130_ANV5_202304271616                                0            3017         inf
ANVIL_eMERGE_HMB_eMERGEseq_20230130_ANV5_202304271619                                            0            2486         inf
ANVIL_eMERGE_HMB_GSO_eMERGEseq_20230130_ANV5_202304271621                                        0            2491         inf
ANVIL_eMERGE_GRU_IRB_eMERGEseq_20230130_ANV5_202304271613                                        0            2432         inf
ANVIL_eMERGE_GRU_IRB_NPU_eMERGEseq_20230130_ANV5_202304271614                                    0            2569         inf
ANVIL_eMERGE_HMB_NPU_eMERGEseq_20230130_ANV5_202304271624                                        0            2488         inf
ANVIL_eMERGE_GRU_eMERGEseq_20230130_ANV5_202304271612                                            0            2971         inf
ANVIL_NIMH_Broad_WGSPD1_McCarroll_Pato_GRU_10XLRGenomes_20230331_ANV5_202304242054               5151         407376       79.08677926616191
ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050               2618         105468       40.285714285714285
ANVIL_CMH_GAFK_GS_linked_read_20221107_ANV5_202304211527                                         312          2541         8.14423076923077
ANVIL_CMH_GAFK_SCATAC_20221107_ANV5_202304211531                                                 640          1280         2.0
```

The two snapshots we observed indexing failures for, "McCarroll_Pato_GRU" and "McCarroll_Braff_DS", are the clear outliers, but it's also strange that there are multiple snapshots with no files at all. I suspect that will cause IT failures later down the line.

The snapshots with ratios in the range of 10:1 to 1:1 are less of a concern, but I'm still curious about how they have more activities than files.

**Comment by dsotirho-ucsc** [comment]
Exclude these snapshots from anvil2:
- ANVIL_NIMH_Broad_WGSPD1_McCarroll_Pato_GRU_10XLRGenomes_20230331_ANV5_202304242054
- ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
I think you overlooked updating hammerbox. And please use `noqa` on the overly long lines:

```diff
Index: deployments/hammerbox/environment.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/deployments/hammerbox/environment.py b/deployments/hammerbox/environment.py
--- a/deployments/hammerbox/environment.py	(revision 58d416d61dcf6a9d0ef690417a57735fa7b1aa2b)
+++ b/deployments/hammerbox/environment.py	(date 1694540589645)
@@ -175,14 +175,8 @@
     mksrc('datarepo-90bab913',
           'ANVIL_NIMH_Broad_ConvergentNeuro_McCarroll_Eggan_Finkel_SMA_DS_WGS_20230109_ANV5_202304242043',
           3),
-    mksrc('datarepo-cddcc424',
-          'ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_10XLRGenomes_20221115_ANV5_202304242050',
-          561),
     mksrc('datarepo-e4eb7641', 'ANVIL_NIMH_Broad_WGSPD1_McCarroll_Braff_DS_WGS_20221115_ANV5_202304242052', 864),
     mksrc('datarepo-f9aef3dc', 'ANVIL_NIMH_Broad_WGSPD1_McCarroll_Escamilla_DS_WGS_20221103_ANV5_202304242049', 85),
-    mksrc('datarepo-75723148',
-          'ANVIL_NIMH_Broad_WGSPD1_McCarroll_Pato_GRU_10XLRGenomes_20230331_ANV5_202304242054',
-          368),
     mksrc('datarepo-7c00e8e5', 'ANVIL_NIMH_Broad_WGSPD1_McCarroll_Pato_GRU_WGS_20221115_ANV5_202304242056', 8084),
     mksrc('datarepo-aca6a582',
           'ANVIL_NIMH_CIRM_FCDI_ConvergentNeuro_McCarroll_Eggan_GRU_Arrays_20230109_ANV5_202304242046',
Index: deployments/anvilprod/environment.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/deployments/anvilprod/environment.py b/deployments/anvilprod/environment.py
--- a/deployments/anvilprod/environment.py	(revision 58d416d61dcf6a9d0ef690417a57735fa7b1aa2b)
+++ b/deployments/anvilprod/environment.py	(date 1694540627408)
@@ -153,9 +153,7 @@
     mksrc('datarepo-12d56848', 'ANVIL_CSER_NCGENES2_GRU_20221208_ANV5_202304271349', 104),
     mksrc('datarepo-8a4d67ef', 'ANVIL_CSER_SouthSeq_GRU_20221208_ANV5_202304271351', 800),
     mksrc('datarepo-bf58b120', 'ANVIL_HPRC_20230721_ANV5_202307241534', 139548),
-    mksrc('datarepo-f622180d',
-          'ANVIL_NIMH_Broad_ConvergentNeuro_McCarroll_Eggan_CIRM_GRU_VillageData_20230109_ANV5_202304242045',
-          71),
+    mksrc('datarepo-f622180d', 'ANVIL_NIMH_Broad_ConvergentNeuro_McCarroll_Eggan_CIRM_GRU_VillageData_20230109_ANV5_202304242045', 71),  # noqa
     mksrc('datarepo-732d1a55',
           'ANVIL_NIMH_Broad_ConvergentNeuro_McCarroll_Eggan_CIRM_GRU_WGS_20230109_ANV5_202304242048',
           445),

```

**Review by hannes-ucsc — APPROVED** [review]

**Comment by achave11-ucsc** [comment]
No failed notifications were found in the queues.

**Comment by dsotirho-ucsc** [comment]
The issue with two snapshots that were omitted from the `anvil2` catalog have been [reported to Nathan at the Broad on Slack](https://ucsc-gi.slack.com/archives/C03TPJS54DC/p1695141154465849?thread_ts=1693341779.811629&cid=C03TPJS54DC).

**Pull request merged**